cafebabel/users/forms.py

```diff
@@ -0,0 +1,50 @@
+from flask_security.forms import LoginForm
+from flask_security.utils import (_datastore, _pwd_context, get_message,
+                                  use_double_hash, hash_password, get_hmac)
+from flask_security.confirmable import requires_confirmation
+
+
+def verify_and_update_password(password, user):
+    """
+    Copy-pasted function from flask_security and disable double_hash for
+    Django passwords.
+    """
+    if (use_double_hash(user.password)
+            and not user.password.startswith('pbkdf2_sha256')):
+        verified = _pwd_context.verify(get_hmac(password), user.password)
+    else:
+        # Try with original password.
+        verified = _pwd_context.verify(password, user.password)
+    if verified and _pwd_context.needs_update(user.password):
+        user.password = hash_password(password)
+        _datastore.put(user)
```

Comment:
I'm (happily) surprised that you can access the datastore directly from flask-security, I thought you would have to retrieve it from the app.

Reply:
This is at least how flask-security deals with it